include/maliput/routing/route_phase.h

```diff
@@ -0,0 +1,174 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2023, Woven Planet. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <optional>
+#include <vector>
+
+#include "maliput/api/lane_data.h"
+#include "maliput/api/regions.h"
+#include "maliput/api/road_network.h"
+#include "maliput/common/maliput_copyable.h"
+#include "maliput/common/maliput_throw.h"
+
+namespace maliput {
+namespace routing {
+
+/// Manages a phase in a complete Route, towards its completion of a Route.
+///
+/// It is composed of a default maliput::api::LaneSRange which is the corridor
+/// the agent traversing the Route should follow. Alternative parallel adjacent
+/// LaneSRanges might be taken when provided.
+///
+/// (To be discussed): we could split the vector of adjacent LaneSRanges into
+/// two dictionaries (adjacent to the left and adjacent to the right).
```

Comment:
It could be easier to consume yeah.

I am also thinking, what about atomizing the route even more. Within `RoutePhase` we have LaneSRanges.
What about creating an abstaction to represent a LaneSRange in a RoutePhase. In a way that we can (similarly to what we have with Segment and Lanes) provide `to_left` and `to_right` methods for moving around the adjacent LaneSRanges in the `RoutePhase`. From an agent's point of view could be useful.

Reply:
That could be a `Path` type, something like:

```C++
class Path {
 public:
  const maliput::api::LaneSRange& lane_s_range() const;
  std::optional<Path> to_left_path() const;
  std::optional<Path> to_right_path() const;
  std::optional<Path> preceding_path() const;
  std::optional<Path> succeeding_path() const;
};
```

The only thing that I see with this type is that we are duplicating the functionality of the `Lane` type without being able to capture multiple preceding and succeeding `Lanes`. If we decided to not offer a proxy for rules, then I would like to follow the same spirit.

Reply:
Please take a look to the proposed solution. It is not in line with my previous comment but I think it offers a simplified version and probably more scalable than the one I proposed.
As a next step, we can provide a `std::vector<maliput::api::LaneSRange> FindLaneSRanges(const maliput::api::LaneSRange lane_s_range, LaneSRangeRelation relation) const;`. That should offer the two way identification of relation between LaneSRanges (for a pair of ranges, and give a range and the relation find those that match the requirement). WDYT?